File: DP1/DP1_1912_continuousSum.cpp
#if 0
# Ive attended the code.plus class (https://code.plus/)
# And I've referred it's codes
# And core parts can be come fully from the class.
#endif


// d[n] : max sum until n
//        max(d[n], d[n-1] + a[n])

#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main()
{
    int n;
    cin >> n;
    vector<int> a(n);
    vector<int> d(n);
    
    for (int i=0; i<n; i++) {
        cin >> a[i];
        d[i] = a[i];
    }
    // O(n)
    for (int i=1; i<n; i++) {
        d[i] = max(d[i], d[i-1]+a[i]);
    }
    cout << *max_element(d.begin(), d.end()) << '\n';
    return 0;
}

File: DP1/DP1_1912_continuousSum.py
#if 0
# Ive attended the code.plus class (https://code.plus/)
# And I've referred it's codes
# And core parts can be come fully from the class.
#endif


n = int(input())
a = list(map(int, input().split()))
d = [0]*n
for i in range(n):
    d[i] = a[i]
    if i == 0:
        continue
    if d[i] < d[i-1] + a[i]:
        d[i] = d[i-1] + a[i]
print(max(d))